From: Laurent Goderre <laurent.goderre@docker.com>
Subject: Remove dependency with CVE and use native function instead
Bug: https://github.com/brix/crypto-js/security/advisories/GHSA-xwcq-pm8m-c4vf
--- /k8s-helper.ts
+++ /k8s-helper.ts
@@ -19,7 +19,7 @@ import {
   V1Pod,
   V1ConfigMap,
 } from '@kubernetes/client-node';
-import crypto from 'crypto-js';
+import * as crypto from 'crypto';
 import * as fs from 'fs';
 import { PartialArgoWorkflow } from './workflow-helper.js';
 import { parseError, findFileOnPodVolume, isAllowedResourceName } from './utils.js';
@@ -61,7 +61,8 @@ const k8sV1CustomObjectClient = kc.makeApiClient(CustomObjectsApi);
 
 function getNameOfViewerResource(logdir: string): string {
   // TODO: find some hash function with shorter resulting message.
-  return 'viewer-' + crypto.SHA1(logdir);
+  const hash = crypto.createHmac('sha1', logdir)
+  return 'viewer-' + hash.digest('hex');
 }
 
 /**
--- a/package.json
+++ b/package.json
@@ -7,7 +7,6 @@
     "@google-cloud/storage": "^2.5.0",
     "@kubernetes/client-node": "^1.4.0",
     "axios": ">=1.13.5",
-    "crypto-js": "^4.2.0",
     "express": "^5.1.0",
     "form-data": "^2.5.4",
     "gunzip-maybe": "^1.4.1",
@@ -20,7 +19,6 @@
   },
   "devDependencies": {
-    "@types/crypto-js": "^3.1.43",
     "@types/express": "^4.11.1",
     "@types/gunzip-maybe": "^1.4.0",
