--- a/Makefile
+++ b/Makefile
@@ -53,7 +53,7 @@ GO_LDFLAGS         := -X $(VPREFIX).Branch=$(GIT_BRANCH) \
 
 UNAME_S := $(shell uname -s)
 ifeq ($(UNAME_S),Linux)
-  GO_FLAGS := -ldflags "-extldflags \"-static\" -s -w $(GO_LDFLAGS)" -tags netgo
+  GO_FLAGS := -buildvcs=false -ldflags "-extldflags \"-static\" -s -w $(GO_LDFLAGS)" -tags netgo
 else
   GO_FLAGS := -ldflags "-s -w $(GO_LDFLAGS)" -tags netgo
 endif
