From 5135ba26f898d492a70b86b29a6e4b9381e1ad63 Mon Sep 17 00:00:00 2001
From: Christian Dupuis <cd@docker.com>
Date: Wed, 28 May 2025 23:09:56 +0200
Subject: [PATCH] Update org.bouncycastle:bcprov-jdk18on:1.78

---
 .../classpath/BuildScriptClasspathIntegrationSpec.groovy        | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy b/subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy
index d0d163c118e..c44ee78f411 100644
--- a/subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy
+++ b/subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy
@@ -311,7 +311,7 @@ class BuildScriptClasspathIntegrationSpec extends AbstractIntegrationSpec implem
             buildscript {
                 ${mavenCentralRepository()}
                 dependencies {
-                    classpath("org.bouncycastle:bcprov-jdk18on:1.77")
+                    classpath("org.bouncycastle:bcprov-jdk18on:1.78")
                 }
             }
 
-- 
2.49.0
