diff --git a/cmake/modules/Distutils.cmake b/cmake/modules/Distutils.cmake
index daaae4b..0a4d0f4 100644
--- a/cmake/modules/Distutils.cmake
+++ b/cmake/modules/Distutils.cmake
@@ -30,9 +30,6 @@ function(distutils_install_module name)
   install(CODE "
     set(options --prefix=${CMAKE_INSTALL_PREFIX})
     if(DEFINED ENV{DESTDIR})
-      if(EXISTS /etc/debian_version)
-        list(APPEND options --install-layout=deb)
-      endif()
       list(APPEND options
         --root=\$ENV{DESTDIR}
         --single-version-externally-managed)
@@ -140,14 +137,11 @@ function(distutils_install_cython_module name)
     set(options
       --prefix=${CMAKE_INSTALL_PREFIX}
       --use-pep517
       --no-build-isolation
       --no-deps
       --ignore-installed)
     if(DEFINED ENV{DESTDIR})
-      if(EXISTS /etc/debian_version)
-        list(APPEND env_vars \"DEB_PYTHON_INSTALL_LAYOUT=deb\")
-      endif()
       list(APPEND options --root=\$ENV{DESTDIR})
     else()
       list(APPEND options --root=/)
     endif()
