diff --git a/cmake/modules/BuildOpentelemetry.cmake b/cmake/modules/BuildOpentelemetry.cmake
--- a/cmake/modules/BuildOpentelemetry.cmake
+++ b/cmake/modules/BuildOpentelemetry.cmake
@@ -12,5 +12,7 @@ function(build_opentelemetry)
   set(opentelemetry_CMAKE_ARGS -DCMAKE_POSITION_INDEPENDENT_CODE=ON
                                -DWITH_JAEGER=ON
+                               -DWITH_ABSEIL=ON
+                               -DCMAKE_CXX_STANDARD=14
                                -DBUILD_TESTING=OFF
                                -DCMAKE_BUILD_TYPE=Release
                                -DWITH_EXAMPLES=OFF)
@@ -18,1 +20,9 @@ function(build_opentelemetry)
+  find_package(absl CONFIG REQUIRED)
+  set(opentelemetry_absl_deps
+      absl::bad_variant_access
+      absl::any
+      absl::base
+      absl::bits
+      absl::city)
+
   set(opentelemetry_libs
@@ -80,5 +90,5 @@ function(build_opentelemetry)
   set_target_properties(
     opentelemetry::libopentelemetry
     PROPERTIES
-      INTERFACE_LINK_LIBRARIES "${opentelemetry_deps}"
+      INTERFACE_LINK_LIBRARIES "${opentelemetry_deps};${opentelemetry_absl_deps}"
       INTERFACE_INCLUDE_DIRECTORIES "${opentelemetry_include_dir}")
