diff --git a/src/jaegertracing/opentelemetry-cpp/exporters/jaeger/src/TUDPTransport.h b/src/jaegertracing/opentelemetry-cpp/exporters/jaeger/src/TUDPTransport.h
--- a/src/jaegertracing/opentelemetry-cpp/exporters/jaeger/src/TUDPTransport.h
+++ b/src/jaegertracing/opentelemetry-cpp/exporters/jaeger/src/TUDPTransport.h
@@ -8,6 +8,7 @@
 #else
 #  include <netdb.h>
 #  include <string.h>
+#  include <unistd.h>
 #  include <sys/socket.h>
 #  include <sys/types.h>
 #endif
