diff --git a/abuild.in b/abuild.in
index 1dfcb1f..d505592 100644
--- a/abuild.in
+++ b/abuild.in
@@ -935,6 +935,19 @@ pre_split() {
 	provides=""
 	install_if=""
 
+	# DHI: handle provides and replaces overwrites from subpackages
+    if [ -n "$subpkgname" ]; then
+		_subpkg_var=$(echo "$subpkgname" | tr -- '-.' '_' | sed 's/^[0-9]*//')
+		_provides_var="${_subpkg_var}_provides"
+		_replaces_var="${_subpkg_var}_replaces"
+		_provides_val=$(eval echo '$'${_provides_var})
+		_replaces_val=$(eval echo '$'${_replaces_var})
+		[ "$_provides_val" = "$_provides_var" ] && _provides_val=""
+		[ "$_replaces_val" = "$_replaces_var" ] && _replaces_val=""
+		[ -n "$_provides_val" ] && provides="$_provides_val"
+		[ -n "$_replaces_val" ] && replaces="$_replaces_val"
+	fi
+
 	cd "$pkgdir"
 }
 
@@ -1796,7 +1809,10 @@ create_apks() {
 			datadir="$pkgbasedir"/$subpkgname
 			scan_shared_objects "$subpkgname" "$dir" "$datadir"
 			scan_symlink_targets "$subpkgname" "$dir" "$datadir"
-			scan_python3_dependency "$subpkgname" "$dir" "$datadir"
+			# DHI: python3 dependency scanning is disabled if PYTHON_VERSION is not set			
+			if [ -z "$PYTHON_VERSION" ]; then
+				scan_python3_dependency "$subpkgname" "$dir" "$datadir"
+			fi
 			scan_debuginfo_targets "$subpkgname" "$dir" "$datadir"
 			scan_pkgconfig_depends "$subpkgname" "$dir" "$datadir"
 		fi
@@ -2186,8 +2202,9 @@ fishcomp() {
 default_pyc() {
 	depends=
 	pkgdesc="Precompiled Python bytecode for ${subpkgname%-pyc}"
-	install_if="${subpkgname%-pyc}=$pkgver-r$pkgrel pyc"
-
+	# DHI: add the python version to the install_if if it is set
+	install_if="${subpkgname%-pyc}=$pkgver-r$pkgrel pyc${PYTHON_VERSION:+-$PYTHON_VERSION}"
+	
 	cd "$pkgdir"
 	local IFS=$'\n'
 	amove $(find usr/lib/python* -type d -name __pycache__)
