--- a/APKBUILD
+++ b/APKBUILD
@@ -120,9 +120,22 @@
 
 	rm -f "$pkgdir"/usr/lib/erlang/lib/$lib-*/src/*.erl
 	amove usr/lib/erlang/lib/$lib-*
+
+	case "$lib" in
+	dialyzer)
+		amove usr/bin/dialyzer usr/bin/typer
+		amove usr/lib/erlang/bin/dialyzer usr/lib/erlang/bin/typer
+		amove usr/lib/erlang/erts-*/bin/dialyzer usr/lib/erlang/erts-*/bin/typer
+		;;
+	esac
 }
 
 dev() {
+	amove usr/lib/erlang/lib/common_test-*
+	amove usr/bin/ct_run
+	amove usr/lib/erlang/bin/ct_run
+	amove usr/lib/erlang/erts-*/bin/ct_run
+
 	default_dev
 	_default_provide
 }
