--- a/production/helm/loki/src/helm-test/canary_test.go	2026-02-26 11:59:09
+++ b/production/helm/loki/src/helm-test/canary_test.go	2026-02-26 08:49:23
@@ -129,7 +129,7 @@ func testResultCanary(t *testing.T, ctx context.Contex
 	body, err := io.ReadAll(rsp.Body)
 	require.NoError(t, err, "Failed to read response body")
 
-	p, err := textparse.New(body, rsp.Header.Get("Content-Type"), true, nil)
+	p, err := textparse.New(body, rsp.Header.Get("Content-Type"), "", true, false, labels.NewSymbolTable())
 	require.NoError(t, err, "Failed to create Prometheus parser")
 
 	for {
