--- build.gradle
+++ build.gradle
@@ -39,7 +39,7 @@
 
 plugins {
   id "de.undercouch.download" version "5.3.0"
-  id 'com.diffplug.spotless' version '6.25.0'
+  id 'com.diffplug.spotless' version '6.25.0' // disabled for DHI build
   id 'checkstyle'
 }
 
@@ -135,19 +135,19 @@
   configDirectory.set(rootProject.file("config/checkstyle/"))
 }
 
-spotless {
-  java {
-    target fileTree('.') {
-      include '**/*.java'
-      exclude '**/build/**', '**/build-*/**'
-    }
-    removeUnusedImports()
-    importOrder()
-    eclipse().withP2Mirrors(Map.of("https://download.eclipse.org/", "https://mirror.umd.edu/eclipse/")).configFile rootProject.file('config/formatterConfig.xml')
-    trimTrailingWhitespace()
-    endWithNewline()
-  }
-}
+// spotless {
+//   java {
+//     target fileTree('.') {
+//       include '**/*.java'
+//       exclude '**/build/**', '**/build-*/**'
+//     }
+//     removeUnusedImports()
+//     importOrder()
+//     eclipse().withP2Mirrors(Map.of("https://download.eclipse.org/", "https://mirror.umd.edu/eclipse/")).configFile rootProject.file('config/formatterConfig.xml')
+//     trimTrailingWhitespace()
+//     endWithNewline()
+//   }
+// }
 
 opensearchplugin {
     name pluginName
@@ -372,7 +372,7 @@
   }
 }
 
-check.dependsOn spotlessCheck
+// check.dependsOn spotlessCheck // disabled for DHI build
 check.dependsOn jacocoTestReport
 
 task integTestWithSecurity(type: RestIntegTestTask) {
