--- build.gradle
+++ build.gradle
@@ -34,7 +34,7 @@
   id 'idea'
   id 'maven-publish'
   id 'jacoco'
-  id 'com.diffplug.spotless' version '6.25.0' apply false
+  id 'com.diffplug.spotless' version '6.25.0' apply false // disabled for DHI build
 }
 
 group = 'org.opensearch'
@@ -45,7 +45,7 @@
 apply plugin: 'opensearch.rest-resources'
 apply plugin: 'opensearch.pluginzip'
 apply plugin: 'opensearch.java-agent'
-apply plugin: 'com.diffplug.spotless'
+// apply plugin: 'com.diffplug.spotless' // disabled for DHI build
 
 idea {
     module {
@@ -211,7 +211,7 @@
 
 tasks.named("check").configure {
   dependsOn(integTest)
-  dependsOn(spotlessCheck)
+  // dependsOn(spotlessCheck) // disabled for DHI build
 }
 
 integTest {
@@ -264,16 +264,16 @@
   ignoreViolations('org.conscrypt.Platform')
 }
 
-plugins.withId('com.diffplug.spotless') {
-  // Configure Spotless only if/when the plugin is applied (e.g., by parent project)
-  spotless {
-    java {
-      removeUnusedImports()
-      importOrder 'java', 'javax', 'org', 'com'
-      eclipse().withP2Mirrors(Map.of("https://download.eclipse.org/", "https://mirror.umd.edu/eclipse/")).configFile rootProject.file('.eclipseformat.xml')
-    }
-  }
-}
+// plugins.withId('com.diffplug.spotless') {
+//   // Configure Spotless only if/when the plugin is applied (e.g., by parent project)
+//   spotless {
+//     java {
+//       removeUnusedImports()
+//       importOrder 'java', 'javax', 'org', 'com'
+//       eclipse().withP2Mirrors(Map.of("https://download.eclipse.org/", "https://mirror.umd.edu/eclipse/")).configFile rootProject.file('.eclipseformat.xml')
+//     }
+//   }
+// }
 
 tasks.named("run").configure {
     // Configure RunTask with the registered integTest cluster
